internal/pkg/cli/deploy/deploy.go

```diff
@@ -1210,8 +1214,13 @@ func validateAppVersionForAlias(appName string, appVersionGetter versionGetter)
 }
 
 func (d *backendSvcDeployer) validateALBRuntime() error {
-	if d.backendMft.RoutingRule.EmptyOrDisabled() || d.backendMft.RoutingRule.Alias.IsEmpty() || !d.env.HasImportedCerts() {
+	switch {
+	case d.backendMft.RoutingRule.EmptyOrDisabled():
 		return nil
+	case !d.backendMft.RoutingRule.Alias.IsEmpty() && !d.env.HasImportedCerts():
+		return fmt.Errorf(`setting "alias" requires an environment with imported certs`)
+	case d.backendMft.RoutingRule.HostedZone != nil && d.backendMft.RoutingRule.Alias.IsEmpty():
+		return fmt.Errorf(`setting "hosted_zone" requires "alias"`)
```

Comment:
Feel like it would be better to use negative sentences for error msgs.

Reply:
see if you like the new error better